packages/nextjs/test/integration/test/browser.js

```diff
@@ -0,0 +1,151 @@
+const fs = require('fs').promises;
+const { createServer } = require('http');
+const { parse } = require('url');
+const path = require('path');
+
+const next = require('next');
+const puppeteer = require('puppeteer');
+
+const {
+  isSentryRequest,
+  isEventRequest,
+  DEBUG_MODE,
+  logDebug,
+  extractEventFromRequest,
+  extractEnvelopeFromRequest,
+  isSessionRequest,
+  isTransactionRequest,
+} = require('./utils');
+const { log } = console;
+
+/**
+ * Usage: node browser.js <filename>
+ *
+ * ENV Variables:
+ * DEBUG=[bool] - enable requests logging
+ * DEBUG_DEPTH=[int] - set logging depth
+ *
+ * Arguments:
+ * <filename> - filter tests based on filename partial match
+ */
+
+const FILES_FILTER = process.argv[2];
+
+(async () => {
+  let scenarios = await fs.readdir(path.resolve(__dirname, './browser'));
+
+  if (FILES_FILTER) {
+    scenarios = scenarios.filter(file => file.toLowerCase().includes(FILES_FILTER));
+  }
+
+  if (scenarios.length === 0) {
+    console.log('No tests founds.');
+    process.exit(0);
+  } else {
+    if (DEBUG_MODE) {
+      scenarios.forEach(s => console.log(`⊙ Tests Found: ${s}`));
+    }
+  }
+
+  // Silence all the unnecessary server noise. We are capturing errors manualy anyway.
+  if (!DEBUG_MODE) {
+    console.log = () => {};
+    console.error = () => {};
+  }
```

Comment:
It seems `browser` and `server` share some code. Is there any way to not repeat it and make it nicer?